Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1002 @@
+        //Turns non-explicit links into clickable links.
+        text = autolinker.link(text);
+
+        //Use a temporary div to manipulate the links
+        //so that they open in a new window.
+        scratchDiv.innerHTML = text;
+        var links = scratchDiv.querySelectorAll('a');
+
+        for (i = 0; i < links.length; i++) {
+            links[i].setAttribute('target', '_blank');
+        }
+
+        var background = Color.WHITE;
+        var foreground = Color.BLACK;
+        var tmp = '<div class="cesium-infoBox-description-lighter" style="';
+        tmp += 'overflow:auto;';
+        tmp += 'word-wrap:break-word;';
+        tmp += 'background-color:' + background.toCssColorString() + ';';
+        tmp += 'color:' + foreground.toCssColorString() + ';';
+        tmp += '">';
+        tmp += scratchDiv.innerHTML + '</div>';
+        scratchDiv.innerHTML = '';
+
+        //return the final HTML as the description.
+        return tmp;
+    }
+
+    function processWpt(dataSource, geometryNode, entityCollection, sourceUri, uriResolver) {
+        var position = readCoordinateFromNode(geometryNode);
+        if (!defined(position)) {
+            throw new DeveloperError('Position Coordinates are required.');
+        }
+
+        var entity = getOrCreateEntity(geometryNode, entityCollection);
+        entity.position = position;
+        // TODO different icon support
+        // var symbol = queryStringValue(geometryNode, 'sym', namespaces.gpx);
+        entity.billboard = createDefaultBillboard(dataSource._proxy, sourceUri, uriResolver);
+
+        var name = queryStringValue(geometryNode, 'name', namespaces.gpx);
+        entity.name = name;
+        entity.label = createDefaultLabel();
+        entity.label.text = name;
+        entity.description = processDescription(geometryNode, entity, uriResolver);
+    }
+
+    //rte represents route - an ordered list of waypoints representing a series of turn points leading to a destination
+    function processRte(dataSource, geometryNode, entityCollection, sourceUri, uriResolver) {
+        var entity = getOrCreateEntity(geometryNode, entityCollection);
+        entity.description = processDescription(geometryNode, entity, uriResolver);
+
+        //a list of wpt
+        var routePoints = queryNodes(geometryNode, 'rtept', namespaces.gpx);
+        var coordinateTuples = new Array(routePoints.length);
+        for (var i = 0; i < routePoints.length; i++) {
+            processWpt(dataSource, routePoints[i], entityCollection, sourceUri, uriResolver);
+            coordinateTuples[i] = readCoordinateFromNode(routePoints[i]);
+        }
+        entity.polyline = createDefaultPolyline();
+        entity.polyline.positions = coordinateTuples;
+    }
+
+    //trk represents a track - an ordered list of points describing a path.
+    function processTrk(dataSource, geometryNode, entityCollection, sourceUri, uriResolver) {
+        var entity = getOrCreateEntity(geometryNode, entityCollection);
```

Comment:
You'll also want to make sure that there's a default billboard for time-dynamic entities, so you see more than just the track.